pyproject.toml

```diff
@@ -7,5 +7,7 @@ requires = ["setuptools", "wheel",
     "numpy ~= 1.20.0 ; python_version == '3.8' and platform_system == 'Darwin' and platform_machine == 'arm64'",
     "numpy ~= 1.18.0 ; python_version == '3.9' and (platform_system != 'Darwin' or platform_machine != 'arm64')",
     "numpy ~= 1.21.0 ; python_version == '3.9' and platform_system == 'Darwin' and platform_machine == 'arm64'",
-    "numpy ~= 1.21.0 ; python_version >= '3.10'",
+    "numpy ~= 1.21.0 ; python_version == '3.10'",
+    "numpy ~= 1.22.0 ; python_version == '3.11'",
+    "numpy >= 1.23.0 ; python_version >= '3.12'",
```

Comment:
In the era of [NEP 29](https://numpy.org/neps/nep-0029-deprecation_policy.html), each version of NumPy is supporting Python versions released in the prior 42 months and dropping support for Python versions older than that.  For this last line, I used `>=` in both places because using `~=` will definitely cause problems for future versions of Python if this file is not updated.